infra/gcp/namespaces/ensure-namespaces.sh

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env bash
+#
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script creates namespaces, applies roles and role bindings and
+# checks if assigned role binding is the only one in the namespace
+# for k8s-infra projects
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+SCRIPT_DIR=$(dirname "${BASH_SOURCE[0]}")
+. "${SCRIPT_DIR}/lib.sh"
+
+role_filename="namespace-user-role.yml"
+role_binding_filename="namespace-user-role-binding.yml"
+role_binding_name_postfix="-namespace-user-binding"
+
+function usage() {
+    echo "usage: $0" > /dev/stderr
+    echo > /dev/stderr
+}
+
+if [ $# != 0 ]; then
+    usage
+    exit 1
+fi
+
+function create_namespace() {
```

Comment:
Sorry to be a pedant, but shell is so easy to make mistakes in...

each function should validate args:

```
function create_namespace() {
    if [ $# != 1 ]; then
        echo "create_namespace(name) requires 1 argument" >&2
        return 1
    fi
    ns="$1"

}
```

Reply:
agree :-)

Reply:
Done